examples/training/object_detection/pascal_voc/faster_rcnn.py

```diff
@@ -63,206 +66,87 @@
-            height, width = tf.unstack(image_shape, axis=-1)
-            max_length = tf.stack([height, width, height, width], axis=-1)
-
-        clipped_boxes = tf.math.maximum(tf.math.minimum(boxes, max_length), 0.0)
-        return clipped_boxes
-
-
-def get_non_empty_box_indices(boxes):
-    # Selects indices if box height or width is 0.
-    height = boxes[:, 2] - boxes[:, 0]
-    width = boxes[:, 3] - boxes[:, 1]
-    indices = tf.where(tf.logical_and(tf.greater(height, 0), tf.greater(width, 0)))
-    return indices[:, 0]
-
-
-def resize_fn(image, gt_boxes, gt_classes):
-    image, image_info = resize_and_crop_image(
-        image, image_size[:2], image_size[:2], 0.8, 1.25
-    )
-    gt_boxes = resize_and_crop_boxes(
-        gt_boxes, image_info[2, :], image_info[1, :], image_info[3, :]
-    )
-    indices = get_non_empty_box_indices(gt_boxes)
-    gt_boxes = tf.gather(gt_boxes, indices)
-    gt_classes = tf.gather(gt_classes, indices)
-    return image, gt_boxes, gt_classes
-
-
-def flip_fn(image, boxes):
-    if tf.random.uniform([], minval=0, maxval=1, dtype=tf.float32) > 0.5:
-        image = tf.image.flip_left_right(image)
-        y1, x1, y2, x2 = tf.split(boxes, num_or_size_splits=4, axis=-1)
-        boxes = tf.concat([y1, 1.0 - x2, y2, 1.0 - x1], axis=-1)
-    return image, boxes
 
-
-def proc_train_fn(bounding_box_format, img_size):
+def unpackage_inputs(bounding_box_format):
     def apply(inputs):
         image = inputs["image"]
         image = tf.cast(image, tf.float32)
-        image = tf.keras.applications.resnet50.preprocess_input(image)
-        gt_boxes = inputs["objects"]["bbox"]
-        image, gt_boxes = flip_fn(image, gt_boxes)
+        gt_boxes = tf.cast(inputs["objects"]["bbox"], tf.float32)
+        gt_classes = tf.cast(inputs["objects"]["label"], tf.float32)
         gt_boxes = keras_cv.bounding_box.convert_format(
             gt_boxes,
             images=image,
             source="rel_yxyx",
             target=bounding_box_format,
         )
-        gt_classes = tf.cast(inputs["objects"]["label"], tf.float32)
-        image, gt_boxes, gt_classes = resize_fn(image, gt_boxes, gt_classes)
-        gt_classes = tf.expand_dims(gt_classes, axis=-1)
-
         return {
             "images": image,
-            "gt_boxes": gt_boxes,
-            "gt_classes": gt_classes,
+            "bounding_boxes": {"boxes": gt_boxes, "classes": gt_classes},
         }
 
     return apply
 
 
-# TODO(tanzhenyu): consider remove padding while reduce function tracing.
-def pad_fn(examples):
-    gt_boxes = examples.pop("gt_boxes")
-    gt_classes = examples.pop("gt_classes")
-    gt_boxes = gt_boxes.to_tensor(default_value=-1.0, shape=[global_batch, 32, 4])
-    gt_classes = gt_classes.to_tensor(default_value=-1.0, shape=[global_batch, 32, 1])
-    return examples["images"], {
-        "boxes": gt_boxes,
-        "classes": gt_classes,
-    }
+def dict_to_tuple(inputs):
```

Comment:
can we modify pad_fn, and remove `dict_to_tuple`?

Reply:
Done